File: Problem2/Problem2.c
#include <stdio.h>
#include <math.h>
#include <time.h>

#define fnMAX 4000000

int fibonacci(int);

int main() {
  clock_t tic = clock();
  int n = 2;
  int fn = 1;
  int suma = 0;

  while (fn < fnMAX) {
    if (fn%2==0) {
      suma += fn;
    }
    n++;
    fn = fibonacci(n);
  }

  clock_t toc = clock();
  printf("Elapsed: %.10f seconds\n", (double)(toc - tic) / CLOCKS_PER_SEC);

  printf("%d\n", suma);
  return 0;
}

int fibonacci(int n) {
  return (1/sqrt(5))*pow(((1 + sqrt(5))/2),n) - (1/sqrt(5))*pow(((1 - sqrt(5))/2),n);
}

File: Problem2/Problem2.py
import math
import time

def fibonacci(n):
    return int((1.0/math.sqrt(5))*((1.0 + math.sqrt(5))/2.0)**n - (1.0/math.sqrt(5))*((1.0 - math.sqrt(5))/2.0)**n)


def main():
    start_time = time.time()

    n = 2
    fn = 1
    suma = 0

    while fn < 4000000:
        if fn%2 == 0:
            suma +=fn
        fn = fibonacci(n)
        if fn > 4000000:
            break
        #print fn
        n = n+1

    print("--- %.15f seconds ---" % (time.time() - start_time))

    print suma

if __name__ == '__main__':
    main()
